sources/packages/backend/apps/api/src/route-controllers/note/_tests_/e2e/note.institutions.controller.getStudentNotes.e2e-spec.ts

```diff
@@ -0,0 +1,238 @@
+  InstitutionTokenTypes,
+} from "../../../../testHelpers";
+
+describe("NoteInstitutionsController(e2e)-getStudentNotes", () => {
+  let app: INestApplication;
+  let collegeFLocation: InstitutionLocation;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+
+    // College F.
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation(collegeF);
+    await authorizeUserTokenForLocation(
+      dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+  });
+
+  it("Should get all note types for a student when student they are available.", async () => {
+    // Arrange
+    // Create new application.
+    const savedApplication = await saveFakeApplication(db.dataSource, {
+      institutionLocation: collegeFLocation,
+    });
+
+    const student = savedApplication.student;
+    const [
+      generalNotes,
+      systemNotes,
+      applicationNotes,
+      designationNotes,
+      overawardNotes,
+      programNotes,
+      restrictionNotes,
+    ] = await saveFakeStudentNotes(
+      db.dataSource,
+      [
+        createFakeNote(NoteType.General),
+        createFakeNote(NoteType.System),
+        createFakeNote(NoteType.Application),
+        createFakeNote(NoteType.Designation),
+        createFakeNote(NoteType.Overaward),
+        createFakeNote(NoteType.Program),
+        createFakeNote(NoteType.Restriction),
+      ],
+      student.id,
+    );
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const endpoint = `/institutions/note/student/${student.id}`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect([
+        noteToApiReturn(restrictionNotes),
+        noteToApiReturn(programNotes),
+        noteToApiReturn(overawardNotes),
+        noteToApiReturn(designationNotes),
+        noteToApiReturn(applicationNotes),
+        noteToApiReturn(systemNotes),
+        noteToApiReturn(generalNotes),
+      ]);
+  });
+
+  it("Should throw NotFoundException when student was not found.", async () => {
+    // Arrange
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    //Act/Assert
+    return request(app.getHttpServer())
+      .get("/institutions/note/student/999999")
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        statusCode: 404,
+        message: "Student not found.",
+        error: "Not Found",
+      });
+  });
+
+  it("Should cause a bad request when a wrong note type is provided as a filter.", async () => {
```

Comment:
with `@HasStudentDataAccess` in place, the student needs to have an application submitted for the institution for this error to happen